src/shared/awsagent/roles.go

```diff
@@ -174,14 +174,19 @@ func (a *Agent) CreateRolesAnywhereProfileForRole(ctx context.Context, role type
 }
 
 // CreateOtterizeIAMRole creates a new IAM role for service, if one doesn't exist yet
-func (a *Agent) CreateOtterizeIAMRole(ctx context.Context, namespaceName string, accountName string, useSoftDeleteStrategy bool) (*types.Role, error) {
+func (a *Agent) CreateOtterizeIAMRole(ctx context.Context, namespaceName string, accountName string, useSoftDeleteStrategy bool, additionalTrustRelationshipEntries ...StatementEntry) (*types.Role, error) {
 	logger := logrus.WithField("namespace", namespaceName).WithField("account", accountName)
 	exists, role, err := a.GetOtterizeRole(ctx, namespaceName, accountName)
 
 	if err != nil {
 		return nil, errors.Wrap(err)
 	}
 
+	trustPolicy, err := a.generateTrustPolicy(namespaceName, accountName, additionalTrustRelationshipEntries...)
```

Comment:
any specific reason why you moved this to before the `if exists`?

Reply:
I forgot to do it here, but I need to take it into account when deciding if the role needs to be updated or not.